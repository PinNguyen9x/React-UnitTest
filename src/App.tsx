import React from "react";
import { Routes, Route } from "react-router-dom";
import { Navigation } from "./components/Navigation";
import { AuthProvider } from "./hooks/useAuth";
import HomePage from "./pages/HomePage";

const App: React.FC = () => {
  return (
    <AuthProvider>
      <Navigation />
      <Routes>
        <Route path="/" element={<HomePage />} />
        <Route path="/about" element={<div>About Page</div>} />
        <Route path="/contact" element={<div>Contact Page</div>} />
        <Route path="/signin" element={<div></div>} />
      </Routes>
    </AuthProvider>
  );
};

export default App;
