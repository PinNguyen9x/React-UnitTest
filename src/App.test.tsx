import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import App from "./App";

const renderWithRouter = (initialEntry = "/") => {
  return render(
    <MemoryRouter initialEntries={[initialEntry]}>
      <App />
    </MemoryRouter>
  );
};

describe("App Component", () => {
  it("renders the home page by default", () => {
    renderWithRouter();
    expect(screen.getByText("Work List App")).toBeInTheDocument();
  });

  it("renders about page on /about route", () => {
    renderWithRouter("/about");
    expect(screen.getByText("About Page")).toBeInTheDocument();
  });

  it("renders navigation component", () => {
    renderWithRouter();
    expect(screen.getByRole("navigation")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = renderWithRouter();
    expect(container).toMatchSnapshot();
  });
});
