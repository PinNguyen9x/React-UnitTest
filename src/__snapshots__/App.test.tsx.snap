// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches snapshot 1`] = `
<div>
  <nav
    class="bg-white shadow-lg"
  >
    <div
      class="max-w-6xl mx-auto px-4"
    >
      <div
        class="flex justify-between items-center h-16"
      >
        <div
          class="flex-shrink-0"
        >
          <a
            class="text-xl font-bold text-gray-800"
            data-discover="true"
            href="/"
          >
            Logo
          </a>
        </div>
        <div
          class="hidden md:block"
        >
          <ul
            class="flex space-x-8"
          >
            <li>
              <a
                class="text-gray-600 hover:text-blue-600 transition duration-300"
                data-discover="true"
                data-testid="home-link"
                href="/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                class="text-gray-600 hover:text-blue-600 transition duration-300"
                data-discover="true"
                data-testid="about-link"
                href="/about"
              >
                About
              </a>
            </li>
            <li>
              <a
                class="text-gray-600 hover:text-blue-600 transition duration-300"
                data-discover="true"
                data-testid="contact-link"
                href="/contact"
              >
                Contact
              </a>
            </li>
          </ul>
        </div>
        <button
          class="bg-blue-600 text-white px-4 py-2 rounded-md hover:bg-blue-700 
                     transition duration-300 focus:outline-none focus:ring-2 
                     focus:ring-blue-500 focus:ring-opacity-50"
          data-testid="about-button"
        >
          Go to About
        </button>
      </div>
    </div>
  </nav>
  <div>
    Home Page
  </div>
</div>
`;
